src/pages/_middleware.ts

```diff
@@ -0,0 +1,27 @@
+import { NextRequest, NextResponse } from 'next/server';
+import domains from '../../newDomains.json';
+
+const domainsObjects = domains.map(([hostname, path]) => ({
+  hostname,
+  pathname: `/${path}`,
+}));
+
+const pathnames = domains.map(d => `/${d[1]}`);
+
+export default function middleware(req: NextRequest): NextResponse | void {
+  const { pathname } = req.nextUrl;
+
+  if (
+    pathnames.includes(pathname) &&
+    !pathname.includes('.') && // exclude all files in the public folder
+    !pathname.endsWith('/api') // exclude all API routes
+  ) {
+    const { hostname } = domainsObjects.find(d => d.pathname === pathname);
```

Comment:
What happens if the object isn't found? This would error...

```
const { hostname } = domainsObjects.find(d => d.pathname === pathname) ?? {};

if (!hostname)
```